user-guide/Cloud_Platform/Catalog/Tutorials/Tutorial_Register_Catalog_Version_GitHub_Actions.md

```diff
@@ -0,0 +1,156 @@
+
+         - name: Use GitHub environment variable to store Catalog ID
+           run: echo "CATALOG_ID=1742495c-9231-4eeb-a56e-1fec8189246e" >> $GITHUB_ENV
+
+         # Install the Skyline DataMiner CICD Packager Tool
+         - name: Install Skyline DataMiner CICD Packager Tool
+           run: dotnet tool install -g Skyline.DataMiner.CICD.Tools.Packager
+
+         # Create the DataMiner Protocol package
+         - name: Create DataMiner Protocol Package
+           run: dataminer-package-create dmprotocol "${{ github.workspace }}" --name catalog_registration_tutorial --output "${{ github.workspace }}/Packages"
+
+         # Prepare the package file and version details
+         - name: Register on Catalog
+           shell: pwsh
+           run: |
+             $file = Get-Item "${{ github.workspace }}/Packages/catalog_registration_tutorial.dmprotocol"
+             $uri = "https://api.dataminer.services/api/key-catalog/v1-0/catalog/${{ env.CATALOG_ID }}/register/version"
+
+             # Define the form data (package file, version, and description)
+             $formData = @{
+               file = Get-Item $file.FullName
+               versionNumber = "${{ env.VERSION }}"
+               versionDescription = "New version registered by GitHub Actions pipeline"
+             }
+
+             # Set up the API request to register the new version in the catalog
+             Invoke-RestMethod -Uri $uri -Method Post -Headers @{ 'Ocp-Apim-Subscription-Key' = "${{ secrets.API_TOKEN }}" } -Form $formData
+   ```
+
+## Step 2: Add GitHub secrets
+
+To securely store sensitive information like the Catalog ID, API token, and version number, you will need to add GitHub secrets:
+
+1. Create an [organization key](xref:Managing_DCP_keys#organization-keys) token to authenticate the register version call from the Catalog API:
+
+   1. In the [Admin app](https://admin.dataminer.services/), under *Organization* in the sidebar on the left, select the *Keys* page.
+
+   1. At the top of the page, click *New Key*.
+
+   1. Configure the key with a label of your choice and the permission *Register catalog items*.
+
+      ![Organization Key](~/user-guide/images/tutorial_catalog_registration_create_org_key.png)
+
+   1. Copy the key so you can use it later.
+
+   > [!IMPORTANT]
+   > You need to have the *Owner* role in order to access/create organization keys. See [Changing the role of a dataminer.services user](xref:Changing_the_role_of_a_DCP_user) for information on how to change a role for a user.
+
+1. In your GitHub repository, go to the *Settings* page.
+
+1. In the sidebar on the left, select *Secrets and variables* > *Actions*, and then click the *New repository secret* button.
+
+   ![New repository secret button](~/user-guide/images/tutorial_catalog_registration_new_secret.png)
+
+1. Specify the name `API_TOKEN`, and add the organization key you copied earlier as the value.
```

Comment:
Which other secrets should be added and where can the user find their value? The title and intro of this step suggest there should be three secrets, but only the API_TOKEN secret is detailed.

Reply:
i adapted it, originally more secrets were used but forgot to adapt this part of the text.